src/pyodide/internal/setupPackages.ts

```diff
@@ -44,6 +45,7 @@ class SitePackagesDir {
       path: '',
       name: '',
       parts: [],
+      reader: null,
```

Comment:
What about setting `reader: EmbeddedPackagesTarReader` and removing the `node.reader ?? EmbeddedPackagesTarReader` logic in `snapshot.ts`? Would that work?

Reply:
Probably, but doesn't matter much. I plan to get rid of `EmbeddedPackagesTarReader` in a follow up PR.